05-Mining-Protocol.md

```diff
@@ -415,7 +426,7 @@ If the `future_job` field is set to False, the client MUST start to mine on the
 | job_id         | U32           | Identifier of the job as provided by NewMiningJob or NewExtendedMiningJob message   |
 +----------------+---------------+-------------------------------------------------------------------------------------+
 | min_ntime      | OPTION[u32]   | Empty if the job is intended for a future SetNewPrevHash message sent on this       |
-|                |               | channel. Filled with min_ntime if the job is intended to the last sent              |
+|                |               | channel. Filled with min_ntime if the job is intended for the last sent             |
```

Comment:
Maybe I'd reformulate the entire description.
Although this is quite ok, I think it sounds a bit weird "If filled with *some value*, it relates to some message [that contains also min_ntime but with different value]".

how about this?
```
Smallest nTime value available for hashing for the new mining job.

Empty value indicates future job that will be activated once SetNewPrevHash message that provides values of new prev_hash and min_ntime.

If the min_ntime value is set, mining job is immediately active and miner must start mining on it as soon as possible. In that case new mining job uses prev_hash from the last received SetNewPrevHash message.

Reply:
the same for the extended job.

Reply:
> Maybe I'd reformulate the entire description. Although this is quite ok, I think it sounds a bit weird "If filled with _some value_, it relates to some message [that contains also min_ntime but with different value]".
> 
> how about this?
> 
> ```
> Smallest nTime value available for hashing for the new mining job.
> 
> Empty value indicates future job that will be activated once SetNewPrevHash message that provides values of new prev_hash and min_ntime.
> 
> If the min_ntime value is set, mining job is immediately active and miner must start mining on it as soon as possible. In that case new mining job uses prev_hash from the last received SetNewPrevHash message.
> ```

I think reformulating the description is a good idea and I also agree it is well written. Maybe I'm not the intended audience because I'm a noob, but I think providing a clear definition of **job** is critically important from the start. As a reader, it's difficult to discern meaning by reference and I was not able to find a definition elsewhere in the [Specification](https://github.com/stratum-mining/sv2-spec).

_Suggestion to add to the terminology section._

**job**: a unit of work for processing. [Source](https://en.wikipedia.org/wiki/Job_(computing)#:~:text=In%20computing%2C%20a%20job%20is,as%20in%20a%20job%20stream).

What are some ways to narrow this definition specifically for mining?

Reply:
Good insight @Bhaney44. It took me some time to understand `future_job` when first reading the spec.

Any suggestions on a good location for this definition?

Reply:
This is what I updated the definition to:
```
Smallest nTime value available for hashing for the new mining job. 
An empty value indicates this is a future job to be activated once a SetNewPrevHash message is received with a matching job_id. This SetNewPrevHash message provides the new prev_hash and min_ntime. 
If the min_ntime value is set, this mining job is active and miner must  start mining on it immediately. In this case, the new mining job uses the  prev_hash from the last received SetNewPrevHash message.  
```

Reply:
At a more fundemental level, I am still not understanding the `future_job` or `min_ntime` variables. I've read through your definitions a few times and I think conceptually, they make it more clear. But, I'm still missing something and it's probably my fault.

I lack a broader knowledge base about the stratum protocol as a whole because I haven't been able to create a mechanism to get started using the protocol, as explained in Issue #25. Without actually being able to use the protocol I am not able to programmatically interact with these variables enough to understand their purpose or meaning.

Reply:
Not your fault at all. I also think this can get confusing. 

I am going to merge this PR in, and then we should open up another PR that better explains these variables.

Reply:
@Bhaney44 are you on our discord? Would love to work with you on improving the spec to make this clearer

Reply:
Thanks! That's very kind of you. I'm not on Discord. But, I'd love to work with you on improving the spec to make it more clear too.

Reply:
@Bhaney44 @rrybarczyk shall we open another follow up issue in specs repo, just to keep a track on what you guys would like to improve? Also @Bhaney44 I think it would be great to join Discord, we have a big SV2 community there.